dm/pkg/shardddl/optimism/lock.go

```diff
@@ -871,3 +714,419 @@ func GetColumnName(lockID, ddl string, tp ast.AlterTableType) (string, error) {
 	}
 	return "", nil
 }
+
+// checkAddDropColumn check for ALTER TABLE ADD/DROP COLUMN statement
+// FOR ADD COLUMN, check whether add column with a different field or add a droped column
+// FOR DROP COLUMN, return the droped column.
+func (l *Lock) checkAddDropColumn(source, schema, table string, ddl string, prevTable, postTable schemacmp.Table) (string, error) {
+	currTable := l.tables[source][schema][table]
+	defer func() {
+		l.tables[source][schema][table] = currTable
+	}()
+
+	l.tables[source][schema][table] = prevTable
+	oldJoined, err := l.joinNormalTables()
+	if err != nil {
+		// nolint:nilerr
+		return "", nil
```

Comment:
just a question why we can mute joined and compared err?

Reply:
This error won't happen because we only call it when no conflict L262